src/main/java/duke/command/DateCommand.java

```diff
@@ -0,0 +1,86 @@
+package duke.command;
+
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.DatedTask;
+import duke.interaction.Ui;
+import duke.util.Storage;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents the Command for the "date" input by the user.
+ * It displays all task items that fall on an input date.
+ *
+ * @author  Hardy Shein
+ * @version 0.1
+ */
+public class DateCommand extends Command {
+
+    private String dateStr;
+
+    public DateCommand() {
+
+    }
+
+    /**
+     * DateCommand constructor.
+     *
+     * @param dateStr of the date user wants to check.
+     */
+    public DateCommand(String dateStr) {
+        this.dateStr = dateStr;
+    }
+
+    /**
+     * Executes Date behaviour of finding all tasks on a given date.
+     *
+     * @param taskList to access collection of tasks.
+     * @param storage to access save-load functionality.
+     */
+    public void execute(TaskList taskList, Storage storage) {
+        String latestResponse = "";
+        try {
+            LocalDate date =  LocalDate.parse(dateStr);
+            Ui.showLine();
+            int count = 0;
+            for (Task t: taskList.getList()) {
+                if (t instanceof DatedTask) {
+                    DatedTask datedTask = (DatedTask)t;
+                    if (datedTask.getDate().toLocalDate().isEqual(date)) {
+                        count++;
+                        latestResponse += count + "." + datedTask.toString() + "\n";
+                        Ui.printWithIndent(count + "." + datedTask.toString());
+                    }
+                }
+            }
```

Comment:
This is a bit arrowhead style code. Think of how we can reduce the level of nesting.